front/pages/api/super-user/workspaces/index.ts

```diff
@@ -0,0 +1,70 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  getSession,
+  getUserFromSession,
+  planForWorkspace,
+} from "@app/lib/auth";
+import { ReturnedAPIErrorType } from "@app/lib/error";
+import { Workspace } from "@app/lib/models";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { WorkspaceType } from "@app/types/user";
+
+export type GetWorkspacesResponseBody = {
+  workspaces: WorkspaceType[];
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<GetWorkspacesResponseBody | ReturnedAPIErrorType>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const user = await getUserFromSession(session);
+
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  if (!user.isDustSuperUser) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const workspaces = await Workspace.findAll();
```

Comment:
If we go for all workspaces then we crucially need paging.

We have thousands of workspaces objects that’s not desirable to load them all if only to preserve our DB

Ideally we would want the upgraded plans… we can get close enough to it by looking at the workspaces that have a plan set, in which case we’re ok to not have paging I think?

Reply:
Thinking more about it I see the logic of having them all so that we can ctrl f to upgrade…

Of course it’s not sustainable but I guess ok for next 2 weeks

Reply:
I'll get only those that have a non-null plan, and below I'll add a search bar to search for a specific workspace by name / ID